samples/oci-apigw-idcs-auth-basic/README.md

```diff
@@ -1,181 +1,204 @@
 
+Note the _IDCS URL_, this is the URL you see in your browser URL bar, copy the IDCS url ( For example: https://idcs-xxxxxxxxxxx.identity.oraclecloud.com/ ), client-id, client-secret and scope these values are provided to the Basicauth function.
 
 ## Review and customize the function
+
 Review the following files in the current folder:
+
 - [pom.xml](./pom.xml) specifies all the dependencies for your function
 - [func.yaml](./func.yaml) that contains metadata about your function and declares properties
 - [src/main/java/com/example/fn/BasicAuth.java](./src/main/java/com/example/fn/BasicAuth.java) which contains the Java code
 
-The name of your function *basicauth* is specified in [func.yaml](./func.yaml).
+The name of your function _basicauth_ is specified in [func.yaml](./func.yaml).
+
+set the following config variables to the values noted while configuring IDCS. The IDCS URL is the token endpoint that returns the access token after validating credentials
 
-set the following variable in "src/main/java/com/example/utils/ResourceServerConfig.java" to the values  noted while configuring IDCS.
 ```
-public static final String CLIENT_ID = "xxxxxxxxxxx";
-public static final String CLIENT_SECRET = "xxxxxxxxx";
-public static final String IDCS_URL = "https://idcs-xxxxxxxx.identity.oraclecloud.com";
+CLIENT_ID = "xxxxxxxxxxx";
+CLIENT_SECRET = "xxxxxxxxx";
+IDCS_URL = "https://idcs-xxxxxxxx.identity.oraclecloud.com/oauth2/v1/token";
 
 //INFORMATION ABOUT THE TARGET APPLICATION
-public static final String SCOPE_AUD = "display-httprequest-infodisplay-httprequest-info";
+SCOPE_AUD = "display-httprequest-infodisplay-httprequest-info";
 ```
 
+For the unit test to run, set the following variables in src/test/java/com/example/fn/BasicAuthTest.java
+
+```
+    private static final String TEST_IDCS_URL = "https://idcs-xxxxxxxx.identity.oraclecloud.com/oauth2/v1/token";
+    private static final String TEST_CLIENT_ID = "xxxxxxxxxxx";
+    private static final String TEST_CLIENT_SECRET = "xxxxxxxxxxx";
+    private static final String TEST_SCOPE_AUD = "display-httprequest-infodisplay-httprequest-info";
+    private static final String TEST_TOKEN = "xxxxxxxxxxx";
+```
 
 ## Deploy the basicauth function
-In Cloud Shell, run the *fn deploy* command to build the function and its dependencies as a Docker image,
+
+In Cloud Shell, run the _fn deploy_ command to build the function and its dependencies as a Docker image,
 push the image to OCIR, and deploy the function to Oracle Functions in your application.
 
 ![user input icon](./images/userinput.png)
+
 ```
 fn -v deploy --app <app-name>
 ```
+
 ## Invoke the basicauth function in cloud shell
-In Cloud Shell, run *fn invoke* command to invoke the deployed function, returns active status as true if the token is valid or else returns false.
+
+In Cloud Shell, run _fn invoke_ command to invoke the deployed function, returns active status as true if the token is valid or else returns false.
```

Comment:
```suggestion
In Cloud Shell, run `fn invoke` to invoke the deployed function. It should return an active status of true if the token is valid or otherwise returns false.
```